### ISSUE TITLE
Neo Store cannot update itself

### ISSUE THREAD

**Issue by Prodigalmanboy** [NeoApplications/Neo-Store#614]
I just started using Neo Store at version 1.00. In the subsequent two new versions, 1.01, and 1.02, I have observed that Neo Store cannot update itself to the newer version. It appears to download, but then simply stops. One has to install F-Droid in order to update Neo Store.

Android 13 on Pixel 8.

This issue is also observed on two other Xiaomi phones I use.

Expected behavior - Neo Store should not require another app to update; it should be able to update itself.

**Comment by Smutley** [comment]
That should be Android 14 on Pixel 8, not Android 13.

Xiaomi phones are Redmi Note 11s, running Lineage OS 13, and Evolution X 13.

**Comment by machiav3lli** [comment]
I guess you have the download stuck or something in that direction (try deleting the item in the downloaded list and sync) … 
Here's some needed information:
- Did you disable battery optimization for NS?
- Is the auto-update enabled?
- Did other apps get updated?
- What is your selected installer?

**Comment by Smutley** [comment]
Thank you for replying.

I did delete the downloaded update; rebooted, with no luck. However, I did not try to re-sync because the new version showed as available (manual install), and or immediately attempted to download and install. No joy.

1) Yes, battery optimization is disabled for Neo Store.
2) It would not update under auto-update, or under manual.
3) Other apps update fine.
4) I only used the root installer on all devices.

**Comment by machiav3lli** [comment]
Can you try installing the latest version through tapping the install button on the suggested version at the bottom of the app sheet?

**Comment by machiav3lli** [comment]
@opusforlife2 using system download manager or the default downloader?

**Comment by Smutley** [comment]
Neo Store still will not itself update 1.02 to 1.03, or 1.04. Still need F-droid to do it.

**Comment by Smutley** [comment]
Nor will Neo Store 1.03 update to 1.04.

**Comment by Smutley** [comment]
I have been messing with this issue on several devices. 

Here is what I found on an Huawei MediaTab T5 Android 9 system that is NOT rooted:

The install program is "looping" ... in other words you tap "install" and it immediately loops the installer back to the prompt to install the update. In frustration I just kept hammering "install" and it eventually installed after about 40+ attempts. Not sure this will help with the issue, but I can repeat it at will on the T5 tablet. To me, it is likely something similar happening in later versions of Android, but you just do not see it if you are using the root installer.

**Comment by yoshimo** [comment]
If you use the new android api, you can only update apps in the background which where installed or updated once with the third party store.
Maybe that causes a problem?

**Comment by yoshimo** [comment]
it is used but I haven't tried older versions of neo store on older OS versions to see when exactly this bug appeared.

**Comment by machiav3lli** [comment]
Yeah, as suggested earlier it's nothing fundamental with the apks, rather maybe some API changes or funny interactions between using the different installers.

So to test the second point: How does it look on the installers' side? Were different installers used in different versions e.g. root till 1.0.1 then session on later? What are the tested installers of this issue?

**Comment by KaKi87** [comment]
I switched to the root installer when it was released, and it wouldn't install the downloaded Neo Store v1.0.4 APK from v1.0.3, I had to install it manually. Thanks

**Comment by Smutley** [comment]
Neither session, nor root installer has worked for me ever, but I only started using Neo Store at 1.01.

I have gone back as far as Android 8 and it it still has the issue. 1.04 will not update to 1.05 unless manually installed, or updated by F-Droid etc.

**Comment by yoshimo** [comment]
I am using the root installer and it can't update.
Just failed again for the 1. 04 - >1.05 Update like the  two previous ones.

**Comment by amano-kenji** [comment]
Is this fixed now? Can neo store update itself automatically now?

**Comment by KaKi87** [comment]
Well, there haven't been an update in a while, so we'll know next time it happens

**Comment by amano-kenji** [comment]
It seems to me that this issue must be solved by operating system. I think GrapheneOS refuses to let neo start update itself for security. If security was so important, then GrapheneOS should let me pick a list of programs that can update themselves.

**Comment by yoshimo** [comment]
How are PlayStore updates handled? It looks like the same problem, a store application updating itself.

**Comment by amano-kenji** [comment]
Simple. Google android trusts google play store and allows google play store to update itself.

Google doesn't let people to extend the trust to third party app stores they trust.

GrapheneOS trusts GrapheneOS app store enough to let it update itself. GrapheneOS doesn't let extend the trust to third party app stores they trust.

Security should not be pursued for its own sake. People should be allowed to take risks.

**Comment by KaKi87** [comment]
It's not a matter of an app trying to update itself. Obtainium can update itself without any issues.

**Comment by machiav3lli** [comment]
Should be fixed since couple of versions
